### ISSUE TITLE
[a11y] UIAccessibilityTraitStaticText need to be added on iOS for text fields

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#41038]
Internal: b/141265252

Running the Google internal a11y checker on iOS on the hello-flutter sample app yields this:

```
Exception Name: GenericFailureException
Exception Reason: Accessibility checks failed.
Exception Details: One or more elements failed the accessibility checks:
 Element: <FlutterSemanticsObject: 0x7fc052577ae0>
   Error: Check "Accessibility label punctuation" failed,
This element has an accessibility label that ends in a period but doesn’t have a text trait. Accessibility labels aren’t sentences and don’t need periods.
If the element visually displays text it should have the UIAccessibilityTraitStaticText trait similar to UITextView or UILabel.

Label found: Button tapped 0 times..
```
